src/components/Input/Input.test.tsx

```diff
@@ -0,0 +1,103 @@
+    const { rerender } = render(<Input {...reqProps} invalidText="Your input is invalid"/>)
+    expect(screen.queryByText(/Your input is invalid/)).not.toBeInTheDocument();
+    rerender(<Input {...reqProps} invalid invalidText="Your input is invalid"/>)
+    expect(screen.queryByText(/Your input is invalid/)).toBeInTheDocument();
+  });
+
+  it('will render a warning message when set as warn', () => {
+    const { rerender } = render(<Input {...reqProps} warnText="Your input is in warning state"/>)
+    expect(screen.queryByText(/Your input is in warning state/)).not.toBeInTheDocument();
+    rerender(<Input {...reqProps} warn warnText="Your input is in warning state"/>)
+    expect(screen.queryByText(/Your input is in warning state/)).toBeInTheDocument();
+  });
+
+  it('will render a react node as the warnText or invalidText', () => {
+    const warnText = (
+      <div>
+        <h1 data-testid="span">The</h1>
+        <p>End Is Near!</p>
+      </div>
+    )
+    const invalidText = (
+      <div>
+        <h1 data-testid="span">This</h1>
+        <p>Is Not Valid!</p>
+      </div>
+    )
+    const {rerender} = render(<Input {...reqProps} warn warnText={warnText} />)
+    expect(screen.queryByText(/The/)).toBeInTheDocument();
+    rerender(<Input {...reqProps} invalid invalidText={invalidText} />)
+    expect(screen.queryByText(/This/)).toBeInTheDocument();
+  });
+
+
+  it('will only render a warning message when input is not invalid', () => {
+    const { rerender } = render(<Input {...reqProps} invalidText="invalid" warn warnText="Your input is in warning state"/>)
+    expect(screen.queryByText(/Your input is in warning state/)).toBeInTheDocument();
+    rerender(<Input {...reqProps} invalid invalidText="invalid" warn warnText="Your input is in warning state"/>)
+    expect(screen.queryByText(/Your input is in warning state/)).not.toBeInTheDocument();
+    expect(screen.queryByText(/invalid/)).toBeInTheDocument();
+  });
+
+  it('will not render a warning message or invalid message when input is disabled or readonly', () => {
+    const { rerender } = render(<Input {...reqProps} disabled invalid invalidText="invalid" warn warnText="Your input is in warning state"/>)
+    expect(screen.queryByText(/Your input is in warning state/)).not.toBeInTheDocument();
+    expect(screen.queryByText(/invalid/)).not.toBeInTheDocument();
+    rerender(<Input {...reqProps} readOnly invalid invalidText="invalid" warn warnText="Your input is in warning state"/>)
+    expect(screen.queryByText(/Your input is in warning state/)).not.toBeInTheDocument();
+    expect(screen.queryByText(/invalid/)).not.toBeInTheDocument();
+  });
+
+  it('will fire the onClick handler when clicked', async () => {
+    const mockedOnClick = jest.fn();
+    render(<Input {...reqProps} onClick={mockedOnClick} />)
+    userEvent.click(screen.getByTestId('test-id'));
+    await waitFor(() => expect(mockedOnClick.mock.calls).toHaveLength(1));
+  });
+});
```

Comment:
I know it doesn't really matter, but trailing linefeeds should be present to suppress github complaining about this. Does VSCode not automatically add them by filetype?

Reply:
It should once I install prettier.